common/src/chain/transaction/partially_signed_transaction.rs

```diff
@@ -0,0 +1,144 @@
+
+use super::{
+    signature::{inputsig::InputWitness, Signable, Transactable},
+    Destination, Transaction, TxOutput,
+};
+use crate::chain::{SignedTransaction, TransactionCreationError, TxInput};
+use serialization::{Decode, Encode};
+use utils::ensure;
+
+#[derive(Debug, Eq, PartialEq, Clone, Encode, Decode)]
+pub struct PartiallySignedTransaction {
+    tx: Transaction,
+    witnesses: Vec<Option<InputWitness>>,
+
+    input_utxos: Vec<Option<TxOutput>>,
+    destinations: Vec<Option<Destination>>,
+}
+
+impl PartiallySignedTransaction {
+    pub fn new(
+        tx: Transaction,
+        witnesses: Vec<Option<InputWitness>>,
+        input_utxos: Vec<Option<TxOutput>>,
+        destinations: Vec<Option<Destination>>,
+    ) -> Result<Self, TransactionCreationError> {
+        ensure!(
+            tx.inputs().len() == witnesses.len(),
+            TransactionCreationError::InvalidWitnessCount
+        );
+
+        ensure!(
+            input_utxos.len() == witnesses.len(),
+            TransactionCreationError::InvalidWitnessCount
+        );
+
+        ensure!(
+            input_utxos.len() == destinations.len(),
+            TransactionCreationError::InvalidWitnessCount
+        );
+
+        Ok(Self {
+            tx,
+            witnesses,
+            input_utxos,
+            destinations,
+        })
+    }
+
+    pub fn new_witnesses(mut self, witnesses: Vec<Option<InputWitness>>) -> Self {
+        self.witnesses = witnesses;
+        self
+    }
+
+    pub fn tx(&self) -> &Transaction {
+        &self.tx
+    }
+
+    pub fn take_tx(self) -> Transaction {
+        self.tx
+    }
+
+    pub fn input_utxos(&self) -> &[Option<TxOutput>] {
+        self.input_utxos.as_ref()
+    }
+
+    pub fn destinations(&self) -> &[Option<Destination>] {
+        self.destinations.as_ref()
+    }
+
+    pub fn witnesses(&self) -> &[Option<InputWitness>] {
+        self.witnesses.as_ref()
+    }
+
+    pub fn count_inputs(&self) -> usize {
+        self.tx.inputs().len()
+    }
+
+    pub fn count_completed_signatures(&self) -> usize {
+        self.witnesses.iter().filter(|w| w.is_some()).count()
+    }
+
+    pub fn all_signatures_available(&self) -> bool {
+        self.witnesses
+            .iter()
+            .enumerate()
+            .zip(&self.destinations)
+            .all(|((_, w), d)| match (w, d) {
+                (Some(InputWitness::NoSignature(_)), None) => true,
+                (Some(InputWitness::NoSignature(_)), Some(_)) => false,
+                (Some(InputWitness::Standard(_)), None) => false,
+                (Some(InputWitness::Standard(_)), Some(_)) => true,
+                (None, _) => false,
+            })
+    }
+
+    pub fn into_signed_tx(self) -> Result<SignedTransaction, TransactionCreationError> {
+        if self.all_signatures_available() {
+            let witnesses = self.witnesses.into_iter().map(|w| w.expect("cannot fail")).collect();
+            Ok(SignedTransaction::new(self.tx, witnesses)?)
```

Comment:
This function used to check if the signatures are valid or not. Is that not possible now?